core/templates/dev/head/components/CollectionSummaryTileDirective.js

```diff
@@ -0,0 +1,52 @@
+// Copyright 2015 The Oppia Authors. All Rights Reserved.
+//
+// Licensed under the Apache License, Version 2.0 (the "License");
+// you may not use this file except in compliance with the License.
+// You may obtain a copy of the License at
+//
+//      http://www.apache.org/licenses/LICENSE-2.0
+//
+// Unless required by applicable law or agreed to in writing, software
+// distributed under the License is distributed on an "AS-IS" BASIS,
+// WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+// See the License for the specific language governing permissions and
+// limitations under the License.
+
+/**
+ * @fileoverview Gallery tile component for collections.
+ *
+ * @author amgowano@stanford.edu (Abraham Mgowano)
+ */
+oppia.directive('collectionSummaryTile', [function() {
+  return {
+    restrict: 'E',
+    scope: {
+      getCollectionId: '&collectionId',
+      getCollectionTitle: '&collectionTitle',
+      getObjective: '&objective',
+      getNumExplorations: '&numExplorations',
+      getLastUpdatedMsec: '&lastUpdatedMsec',
+      getThumbnailIconUrl: '&thumbnailIconUrl',
+      getThumbnailBgColor: '&thumbnailBgColor'
+    },
+    templateUrl: 'summaryTile/collection',
+    controller: [
+      '$scope', 'oppiaDatetimeFormatter',
+      function($scope, oppiaDatetimeFormatter) {
+        $scope.getLastUpdatedDatetime = function() {
+          return oppiaDatetimeFormatter.getLocaleAbbreviatedDatetimeString(
+            $scope.getLastUpdatedMsec());
+        };
+
+        $scope.wasRecentlyUpdated = function() {
+          return oppiaDatetimeFormatter.isRecent($scope.getLastUpdatedMsec());
+        };
+
+        $scope.getCollectionLink = function() {
+          var result = '/collection/' + $scope.getCollectionId();
```

Comment:
Prefer to use ``UrlInterpolationService`` here. Similar to how in CollectionPlayer.js there is a constant defined for ``COLLECTION_DATA_URL_TEMPLATE``, you can define a template URL constant here with the value ``/collection/<collection_id>``, inject it, then use ``UrlInterpolationService`` to build the URL returned here. This is to protect against potential XSS exploits. Only ``UrlInterpolationService`` should be used to build URLs.